src/reason_parser.mly

```diff
@@ -2490,6 +2520,140 @@ labeled_simple_pattern:
+      {
+        (* Foo></Foo *)
+        let (component, start) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        let _ = ensureTagsAreEqual start $3 loc in
+        component [("", mktailexp_extension loc [] None)] ~loc
+      }
+  | jsx_start_tag_body GREATERLESS jsx_tag_siblings jsx_name
+      {
+        let (component, start) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        let _ = ensureTagsAreEqual start $4 loc in
+        component [("", mktailexp_extension loc $3 None)] ~loc
+      }
+  | jsx_start_tag_body GREATER LESS jsx_tag_siblings jsx_name
+      {
+        let (component, start) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        let _ = ensureTagsAreEqual start $5 loc in
+        component [("", mktailexp_extension loc $4 None)] ~loc
+      }
+  | jsx_start_tag_body GREATER jsx_siblings jsx_name
+      {
+        (* Foo> <Tags /> </Foo *)
+        let (component, start) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        let _ = ensureTagsAreEqual start $4 loc in
+        let siblings =
+          if List.length $3 > 0 then
+            $3
+          else
+            []
+        in
+        component [("", mktailexp_extension loc siblings None)] ~loc
+      }
+;
+
+jsx_tag:
+  | LESS jsx_start_tag_body SLASHGREATER
+      {
+        (* Foo /> *)
+        let (component, _) = $2 in
+        let loc = mklocation $symbolstartpos $endpos in
+        component [("", mktailexp_extension loc [] None)] ~loc
+      }
+  | LESS jsx_separate_open_close GREATER
+      {
+        (* /Foo> *)
+        $2
+      }
+;
+
+jsx_tag_siblings:
+  | jsx_start_tag_body SLASHGREATERLESSSLASH
+      {
+        let (component, _) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        [component [("", mktailexp_extension loc [] None)] ~loc]
+      }
+  | jsx_start_tag_body SLASHGREATERLESS jsx_tag_siblings
+      {
+        (* Tag /><Tag *)
+        let (component, _) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        [component [("", mktailexp_extension loc [] None)] ~loc] @ $3
+      }
+  | jsx_start_tag_body SLASHGREATER jsx_siblings
+      {
+        (* Tag /> <Tags *)
+        let (component, _) = $1 in
+        let loc = mklocation $symbolstartpos $endpos in
+        [component [("", mktailexp_extension loc [] None)] ~loc] @ $3
+      }
+  | jsx_separate_open_close GREATERLESSSLASH
+      {
+        [$1]
+      }
+  | jsx_separate_open_close GREATER jsx_siblings
+      {
+        (* Tag> <Tags *)
+        [$1] @ $3
+      }
+  | jsx_separate_open_close GREATERLESS jsx_tag_siblings
+      {
+        (* Tag><Tags *)
+        [$1] @ $3
+     }
+;
+
+jsx_siblings:
+    LESSSLASH { [] }
+  | STRING jsx_siblings
+      {
+        let (s, d) = $1 in
+        [mkexp (Pexp_constant (Const_string (s, d)))] @ $2
+      }
+  | LESS jsx_tag_siblings
+      {
+        $2
+      }
+  | LBRACE expr RBRACE jsx_siblings
```

Comment:
Something is incorrect here.
If you try `let a = <Foo a>{1}</Foo>;`, you will notice this is not idempotent.